system/lib/pthread/emscripten_yield.c

```diff
@@ -41,6 +37,9 @@ void _emscripten_yield() {
     // singlethreaded.
     emscripten_main_thread_process_queued_calls();
```

Comment:
In another comment you said that `emscripten_yield` only does dlopen operations, but it also calls `emscripten_main_thread_process_queued_calls` - ? That is the issue I worry about. Making futex_wait call a dlopen method seems much safer to me compared to also calling the main proxied calls logic, where anything can be waiting.

Reply:
This code is inside the `is_runtime_thread` check.. so my comment should, more accurately, have read that calling _emscripten_yield from a background thread will only process the dlopen queue.

I agree that this change should definitely not result in workers processing the system proxy queue during `futex_wait`.   IIRC only the main thread should do that for historical reasons.